src/hooks/usePropsGetters/useAddToFavoritesButtonProps.ts

```diff
@@ -14,13 +16,16 @@ export default function useAddToFavoritesButtonProps(
     [favorited]
   );
 
+  const isResultFavorited = (result: Partial<QuizResultData>) =>
+    result?.data?.id && favoriteItems?.includes(result?.data?.id);
+
   const getAddToFavoritesButtonProps: GetAddToFavoritesButtonProps = useCallback(
     (result, price) => ({
       className: `${'cio-result-card-favorites-button'} ${
-        favorited?.[result?.data?.id || '0'] ? 'favorited' : ''
+        result?.data?.id && favoriteItems?.includes(result?.data?.id) ? 'favorited' : ''
```

Comment:
If I'm understanding this correctly, the button's state doesn't depend on the local `favorited` state anymore. Since we are giving the responsibility of that to the consumer and using the passed `favoriteItems` for that purpose, can we get rid of `const [favorited, setFavorited] = useState<Favorited>({});`?